content/push/publish.textile

```diff
@@ -0,0 +1,768 @@
+```
+
+The "@DeviceDetails@":/api/realtime-sdk/push-admin object contains the @deviceId@, @clientId@, along with other recipient attributes, and it can be accessed once the device is registered.
+
+h3(#sub-clientID). Subscibe to channels using @clientId@
+
+A @clientId@ is used to identify a specific client within a system. This client can be an individual user, a device, or even a group of users or devices that share common permissions or settings.
+
+The @AblyRealtime.push.activate@ method automatically links a @deviceID@ with the @clientId@ from the @AblyRealtime@ instance.
+
+Once a device is associated with a @clientId@, you can subscribe to push notifications for all devices linked to that @clientId@ with a single operation. For instance, if you have multiple devices (like a phone, tablet, and laptop) associated with the same @clientId@, you can efficiently manage their push notification subscriptions together.
+
+On the other hand, if a user no longer uses a device, it is automatically disassociated from the @clientId@, resulting in the loss of its subscription to the channels associated with that @clientId@.
+
+
+The following is an example of subscribing to receive push notifications on a channel using @clientId@:
+
+```[android]
+realtime.channels.get("pushenabled:foo").push.subscribeClient();
+
+// or
+
+realtime.channels.get("pushenabled:foo").push.subscribeClientAsync(new CompletionListener() {
+    @Override
+    public void onSuccess() {}
+
+    @Override
+    public void onError(ErrorInfo errorInfo) {
+        // Handle error.
+    }
+});
+```
+
+```[swift]
+realtime.channels.get("pushenabled:foo").push.subscribeClient { error
+    // Check error.
+}
+```
+
+```[objc]
+[[realtime.channels get:@"pushenabled:foo"].push subscribeClient:^(ARTErrorInfo *error) {
+    // Check error.
+}];
+```
+
+Publishing is the process that sends push notifications from the source the client application or server to the end-user's device. Ably supports two methods of publishing: Publish directly and publish via channels.
+
+"Publishing directly":#direct-publishing sends notifications directly to individual devices without channels. Each notification targets a specific device, akin to sending a personal message. This method is useful for sending personalized or specific notifications to individual users.
+
+"Publishing via channles":#channel-pub automatically sends push notifications to multiple devices through channels. Ably ensures that messages sent to a channel are delivered to all devices subscribed to it. This method is beneficial for sending the same notification to a group of devices, such as users subscribed to a particular topic or group.
+
+Both of these methods use the "push notification payload structure":#payload-structure, which includes the notification's content and any additional data required by the application. You can customize this payload structure for iOS and Android platforms to enhance the user experience.
```

Comment:
I think you're missing a section here which is how to then publish a push notification that will be sent to those subscribed to a channel.

The content here looks like it's supposed to be in different sections or is repeated from elsewhere.